File: workers/my-python-worker/src/worker.py
# src/worker.py
import asyncio
from datetime import timedelta
from temporalio import activity, workflow
from temporalio.client import Client
from temporalio.worker import Worker
import os

@activity.defn
async def my_activity(input: str) -> dict:
    """Activity - can have side effects"""
    activity.logger.info(f"Processing: {input}")
    return {"success": True, "message": f"Processed {input}"}


@workflow.defn
class MyWorkflow:
    @workflow.run
    async def run(self, data: str) -> str:
        result = await workflow.execute_activity(
            my_activity,
            data,
            start_to_close_timeout=timedelta(minutes=10),
            retry_policy=workflow.RetryPolicy(
                initial_interval=timedelta(seconds=30),
                backoff_coefficient=2,
                maximum_attempts=3,
            ),
        )
        if not result["success"]:
            raise Exception(f"Activity failed: {result['message']}")
        return result["message"]


async def main():
    client = await Client.connect(
        os.getenv("TEMPORAL_ADDRESS", "localhost:7233")
    )
    worker = Worker(
        client,
        task_queue=os.getenv("TASK_QUEUE", "my-tasks"),
        workflows=[MyWorkflow],
        activities=[my_activity],
    )
    print(f"Worker started on queue: {worker.task_queue}")
    await worker.run()


if __name__ == "__main__":
    asyncio.run(main())

File: ssmd-worker/src/worker.ts
import { NativeConnection, Worker } from '@temporalio/worker';
import * as activities from './activities';

async function run() {
  const address = process.env.TEMPORAL_ADDRESS || 'localhost:7233';
  const namespace = process.env.TEMPORAL_NAMESPACE || 'default';
  const taskQueue = process.env.TASK_QUEUE || 'ssmd-tasks';

  console.log('Connecting to Temporal at ' + address);
  console.log('Namespace: ' + namespace);
  console.log('Task queue: ' + taskQueue);

  const connection = await NativeConnection.connect({ address });

  const worker = await Worker.create({
    connection,
    namespace,
    taskQueue,
    workflowsPath: require.resolve('./workflows'),
    activities,
  });

  console.log('Worker started, waiting for tasks...');
  await worker.run();
}

run().catch((err) => {
  console.error('Worker failed:', err);
  process.exit(1);
});
